doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -384,297 +381,248 @@ Rewriting with definitional equality
-   compilation cost is higher, so it is worth using only for intensive
-   computations. Depending on the configuration, this tactic can either default to
-   :tacn:`vm_compute`, recompile dependencies or fail due to some missing
-   precompiled dependencies,
-   see :ref:`the native-compiler option <native-compiler-options>` for details.
-
-   .. flag:: NativeCompute Timing
-
-      This flag causes all calls to the native compiler to print
-      timing information for the conversion to native code,
-      compilation, execution, and reification phases of native
-      compilation.  Timing is printed in units of seconds of
-      wall-clock time.
-
-   .. flag:: NativeCompute Profiling
-
-      On Linux, if you have the ``perf`` profiler installed, this flag makes
-      it possible to profile :tacn:`native_compute` evaluations.
-
-   .. opt:: NativeCompute Profile Filename @string
-      :name: NativeCompute Profile Filename
-
-      This option specifies the profile output; the default is
-      ``native_compute_profile.data``. The actual filename used
-      will contain extra characters to avoid overwriting an existing file; that
-      filename is reported to the user.
-      That means you can individually profile multiple uses of
-      :tacn:`native_compute` in a script. From the Linux command line, run ``perf report``
-      on the profile file to see the results. Consult the ``perf`` documentation
-      for more details.
+      A variant form of :tacn:`cbv`.
 
    :opt:`Debug` ``"Cbv"`` makes :tacn:`cbv` (and its derivative :tacn:`compute`) print
    information about the constants it encounters and the unfolding decisions it
    makes.
 
-.. tacn:: red
-   :name: red
+.. tacn:: red @simple_occurrences
 
-   This tactic applies to a goal that has the form::
+   βιζ-reduces the constant at the head of `T` (which may be called
+   the :gdef:`head constant`; :gdef:`head` means the beginning
+   of the term), if possible,
+   in the selected hypotheses and/or the goal, which must have the form:
 
-     forall (x:T1) ... (xk:Tk), T
+     :n:`{? forall @open_binders,} T`
 
-   with :g:`T` :math:`\beta`:math:`\iota`:math:`\zeta`-reducing to :g:`c t`:sub:`1` :g:`... t`:sub:`n` and :g:`c` a
-   constant. If :g:`c` is transparent then it replaces :g:`c` with its
-   definition (say :g:`t`) and then reduces
-   :g:`(t t`:sub:`1` :g:`... t`:sub:`n` :g:`)` according to :math:`\beta`:math:`\iota`:math:`\zeta`-reduction rules.
+   (where `T` does not begin with a `forall`) to :n:`c t__1 … t__n`
+   where :g:`c` is a constant.
+   If :g:`c` is transparent then it replaces :g:`c` with its
+   definition and reduces again until no further reduction is possible.
 
-.. exn:: Not reducible.
-   :undocumented:
+   .. exn:: Not reducible.
+      :undocumented:
 
-.. exn:: No head constant to reduce.
-   :undocumented:
+   .. exn:: No head constant to reduce.
+      :undocumented:
 
-.. tacn:: hnf
-   :name: hnf
+.. tacn:: hnf @simple_occurrences
```

Comment:
Perhaps it would be better to describe `hnf` *after* `simpl` and `cbn` since it shares most of its implementation and is described by contrasting it to `simpl` and `cbn`.